bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/AccountHandler.java

```diff
@@ -792,42 +812,50 @@ private void handlePushDeviceCommand(DopplerId dopplerId, String command, String
-        }
-
-        Connection connection = this.connection;
-        if (connection == null || !connection.getIsLoggedIn()) {
-            return;
-        }
-        Activity[] activities = connection.getActivities(10, pushActivity.timestamp);
-        Activity currentActivity = null;
-        String search = key.registeredUserId + "#" + key.entryId;
-        for (Activity activity : activities) {
-            if (StringUtils.equals(activity.id, search)) {
-                currentActivity = activity;
-                break;
-            }
-        }
-        if (currentActivity == null) {
-            return;
+        pushActivityQueue.add(payload);
+        if (pushActivityQueueRunning.compareAndSet(false, true)) {
+            queuedPushActivity();
         }
+    }
 
+    private void queuedPushActivity() {
         @Nullable
-        SourceDeviceId @Nullable [] sourceDeviceIds = currentActivity.sourceDeviceIds;
-        if (sourceDeviceIds != null) {
-            for (SourceDeviceId sourceDeviceId : sourceDeviceIds) {
-                if (sourceDeviceId != null) {
-                    EchoHandler echoHandler = findEchoHandlerBySerialNumber(sourceDeviceId.serialNumber);
-                    if (echoHandler != null) {
-                        echoHandler.handlePushActivity(currentActivity);
-                    }
+        String payload = pushActivityQueue.poll();
+        if (payload != null) {
+            try {
+                JsonCommandPayloadPushActivity pushActivity = gson.fromJson(payload,
+                        JsonCommandPayloadPushActivity.class);
+
+                Key key = pushActivity.key;
+                if (key == null) {
+                    return;
+                }
+
+                Connection connection = this.connection;
+                if (connection == null || !connection.getIsLoggedIn()) {
+                    return;
                 }
+
+                String search = key.registeredUserId + "#" + key.entryId;
+                Arrays.stream(connection.getActivities(10, pushActivity.timestamp))
+                        .filter(activity -> activity != null && search.equals(activity.id)).findFirst()
+                        .ifPresent(currentActivity -> {
+                            SourceDeviceId[] sourceDeviceIds = currentActivity.sourceDeviceIds;
+                            if (sourceDeviceIds != null) {
+                                Arrays.stream(sourceDeviceIds).filter(Objects::nonNull).map(
+                                        sourceDeviceId -> findEchoHandlerBySerialNumber(sourceDeviceId.serialNumber))
+                                        .filter(Objects::nonNull)
+                                        .forEach(echoHandler -> echoHandler.handlePushActivity(currentActivity));
+                            }
+                        });
+            } finally {
+                pushActivitySenderUnblockFuture = scheduler.schedule(this::queuedPushActivity, 5, TimeUnit.SECONDS);
```

Comment:
Since this is synchronous anyway, no need to unblock, just call `queuedPushActivity`.